core/dbt/parser/manifest.py

```diff
@@ -119,6 +120,10 @@ def __init__(
             root_project, all_projects,
         )
         self._loaded_file_cache: Dict[str, FileBlock] = {}
+        self._perf_logging: bool = False
+        partial_parse = self._partial_parse_enabled()
+        self._perf_info: Dict[str, Any] = {
```

Comment:
Do you think you can create a serializable dataclass for this? something along the lines of:

```python
@dataclass
class ParsingInfo(JsonSchemaMixin, Writable):
    path_count: int = 0
    ...
```

`Writable` is totally optional, it is just a friendlier API to write to disk.

I'm thinking about our future selves whenever I suggest using dataclasses and better types :)

Reply:
I don't think we need JsonSchemaMixin since this is being created internally and we don't need to validate.

Reply:
Yep, totally agree. `JsonSchemaMixin` just makes it easy to serialize the dataclass to a python dict. By default, `JsonSchemaMixin` does not validate data when converting dataclass to a JSON encodable dict.

Reply:
So use to_dict instead of just json.dumps. It's not actually necessary right now since json.dumps works fine, but I suppose we might add some datatypes that use it.

Reply:
Up to you! If it works that works for me. If you do decide to use my suggestion with `JsonSchemaMixin`, you'll want to use `to_dict` with `json.dumps`
```python
write_file(path, json.dumps(self.loader._perf_info.to_dict(), cls=dbt.utils.JSONEncoder, indent=4))
```

Reply:
I spent a bunch of time trying to create some JsonSchemaMixin classes for that (it requires about 4, I think) and could not get to_dict() to work properly. The last complaint was that there was no to_dict() method on lists. In theory I think setting the dataclasses up might be a good idea, but I'm going to stop now, and just use the original plain dicts and lists.